iceoryx_binding_c/source/c_subscriber.cpp

```diff
@@ -39,21 +40,19 @@ iox_sub_t iox_sub_init(iox_sub_storage_t* self,
                        const char* const service,
                        const char* const instance,
                        const char* const event,
-                       const uint64_t queueCapacity,
-                       const uint64_t historyRequest,
-                       const char* const nodeName)
+                       const iox_subscriber_options_t options)
 {
     new (self) cpp2c_Subscriber();
     iox_sub_t me = reinterpret_cast<iox_sub_t>(self);
-    SubscriberOptions options;
-    options.queueCapacity = queueCapacity;
-    options.historyRequest = historyRequest;
-    options.nodeName = NodeName_t(TruncateToCapacity, nodeName);
+    SubscriberOptions subscriberOptions;
+    subscriberOptions.queueCapacity = options.queueCapacity;
+    subscriberOptions.historyRequest = options.historyRequest;
+    subscriberOptions.nodeName = NodeName_t(TruncateToCapacity, options.nodeName);
```

Comment:
Maybe we should print a warning here when options.nodeName is longer then the NodeName_t capacity so that the user is at least informed and does not report a bug that his node name is truncated.

Reply:
cxx::string does this. As stated above I would not add the additional check.